src/components/pages/deals/Deals.js

```diff
@@ -3,31 +3,91 @@ import React, { useState, useEffect } from 'react'
 import itemsServices from '../../../services/items'
 import ProductsOnDeals from '../../products/Products'
 import SearchBar from '../../searchbar/Searchbar'
+import PaginationButton from '../../PaginationButtons/PaginationButton'
 
 import './deals.css'
 
+const PAGE_SIZE = 6
+
 const Deals = () => {
     const [deals, setDeals] = useState([])
     const [isLoading, setIsLoading] = useState(true)
+    const [noDeals, setNoDeals] = useState(false)
+    const [number_of_pages, setNumberOfPages] = useState(null)
+    const [page, setPage] = useState(1)
 
     useEffect(() => {
         const fetch_deals = async () => {
-            const query = 'isOnSale=true'
+            // Get all deals to calculate the number of pages for pagination
+            const all_deals = await itemsServices.getDeals('isOnSale=true')
+            // set number of pages
+            setNumberOfPages(Math.ceil(all_deals.items.length/PAGE_SIZE))
+            // Query made to look for deals and from a certain number
+            const query = `isOnSale=true&from=${(page - 1) * PAGE_SIZE}&size=${PAGE_SIZE}`
             const response_data = await itemsServices.getDeals(query)
             setDeals(response_data.items)
             setIsLoading(false)
         }
         fetch_deals()
-    }, [])
+    }, [page])
+
+    // called when a user does a search with the searchBar component
+    const handleSearch = async (searchItem) => {
+        if (searchItem === '') {
+            setPage(1)
+            setNoDeals(false)
+        } else {
+            // construct a query from the search item
+            const query = `q=${searchItem}`
+            const response = await itemsServices.getSearchedItem(query)
+            //Sibce the backend responds with only 1 item, make an array from that
+            const item = [response.items[0]]
+            if (response.items[0] === undefined) {
+                setNoDeals(true)
+            } else {
+                setDeals(item);
+            }
+        }
+        
+    }
+
+    // Increment the page state when next button is clicked
+    const handleNextPage = () => {
```

Comment:
See if you can move these functions to the Pagination component to avoid repeating them in the Home and Deals pages.